### PR TITLE
Week 2

### PR DESCRIPTION
**Pull request by Erkan-5Minds**

### CODE REVIEW

**Review by stefanluth — COMMENTED** [review]
Try to remove the `.DS_Store` & `Week_2_Arrays/.vscode/c_cpp_properties.json` files :)

You requested a review but you didn't upload any code 👀

**Comment by DevBastiS** [comment]
The title is still WIP (Work in Progress) so I am not sure if I should review here.

File: Week_2_Arrays/caesar/caesar.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                printf("%c", (plaintext[j] - 'a' + k ) % 26 + 'a');
+            }
+            
+            else
+            {
+                printf("%c", plaintext[j]);
+            }
+        }
+        printf("\n");
```
Comment (MKrips) [inline Week_2_Arrays/caesar/caesar.c]: This should be indented 4 spaces, not 8.

File: Week_2_Arrays/readability/readability.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#include <ctype.h>
+#include <string.h>
+
+int main(void) 
+{
+    string text = get_string("Text: ");
+
+    int letters = 0;
+    int words = 0; 
+    int senteces = 0;
```
Comment (MKrips) [inline Week_2_Arrays/readability/readability.c]:
```suggestion
    int sentences = 0;
```

**Review by MKrips — CHANGES_REQUESTED** [review]
Remember to do what Stefan requested.

File: Week_2_Arrays/caesar/caesar.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        printf("Usage: ./caesar key");
+        return 1;
+    }
+
+    for (int i = 0; i < strlen (argv[1]); i++)
+    {
+        if (!isdigit(argv[1][i]))
+        printf("usage: ./caesar key");
+    }
```
Comment (DevBastiS) [inline Week_2_Arrays/caesar/caesar.c]:
```suggestion
    }
    
```

File: Week_2_Arrays/caesar/caesar.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        if (isupper(plaintext[j]))
+        {
+            printf("%c", (plaintext[j] - 'A' + k ) % 26 + 'A');
+        }
+        else if (islower(plaintext[j]))
+        {
+            printf("%c", (plaintext[j] - 'a' + k ) % 26 + 'a');
+        }
+
```
Comment (DevBastiS) [inline Week_2_Arrays/caesar/caesar.c]:
```suggestion
```

File: Week_2_Arrays/readability/readability.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    float L = (float)letters / (float) words * 100;
+    float S = (float)sentences / (float)words * 100;
+
+    int index = (0.0588 * L - 0.296 * S - 15.8);
+
+    if (index < 1)
+    {
+        printf("Before Grade 1\n");
+    }
+
```
Comment (DevBastiS) [inline Week_2_Arrays/readability/readability.c]:
```suggestion
```

File: Week_2_Arrays/readability/readability.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    if (index < 1)
+    {
+        printf("Before Grade 1\n");
+    }
+    
+    else if (index > 16)
+    {
+        printf("Grade 16+\n");
+    }
+
```
Comment (DevBastiS) [inline Week_2_Arrays/readability/readability.c]:
```suggestion
```

File: Week_2_Arrays/scrabble/scrabble.c [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    int score1 = compute_score(word1);
+    int score2 = compute_score(word2);
+
+    if (score1 > score2) 
+    {
+        printf("Player 1 Wins!\n");
+
+        return 0;
+    }
```
Comment (DevBastiS) [inline Week_2_Arrays/scrabble/scrabble.c:23]:
```suggestion
    }

```

File: Week_2_Arrays/scrabble/scrabble.c [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    if (score2 > score1) 
+    {
+        printf("Player 2 Wins!\n");
+
+        return 0; 
+    }
+    else {
+        printf("Tie!\n");
+
+    return 0;
```
Comment (DevBastiS) [inline Week_2_Arrays/scrabble/scrabble.c]:
```suggestion
        return 0;
```

File: Week_2_Arrays/substitution/substitution.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+#include <string.h>
+#include <ctype.h>
+bool is_valid_key(string s);
+int main(int argc, string argv[])
+{
+    if (argc != 2)
+    {
+        printf("Usage: ./substitution key \n");
+        return 1;
+    }
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:13]:
```suggestion
    }
    
```

File: Week_2_Arrays/substitution/substitution.c [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    if (!is_valid_key(argv[1]))
+    {
+        printf("Key must contain 26 characters.\n");
+        return 1;
+    }
+    
+    string s = get_string("plaintext:  ");
+    string differance = argv[1];
+    printf("ciphertext: ");
+    for (int i = 'A'; i < 'Z'; i++)
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:25]: Always use {} notation

File: Week_2_Arrays/substitution/substitution.c [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        differance[i - 'A'] = toupper(differance[i - 'A']) - i;
+    for (int i = 0; i < strlen(s); i++)
+    {
+       if (isalpha(s[i]))
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:32]: Always use {} notation

File: Week_2_Arrays/substitution/substitution.c [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            s[i] = s[i] + differance[s[i]] - (isupper(s[i]) ? 'A' : 'a');
+        printf("%c", s[i]);
+    }
+    printf("\n");
+}
+
+bool is_valid_key(string s)
+{
+    int len = strlen(s);
+    if (len != 26)
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:44]: Always use {} notation

File: Week_2_Arrays/substitution/substitution.c [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        return false;
+    
+    int freq[26] = {0};
+    for (int i = 0; i < len; i++)
+    {   
+        if (!isalpha(s[i]))
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:52]: Always use {} notation

File: Week_2_Arrays/substitution/substitution.c [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+            return false;
+        int index = toupper(s[i]) - 'A';
+        if (freq[index] > 0)
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:59]: Always use {} notation

File: Week_2_Arrays/substitution/substitution.c [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+            return false;
+        freq[index]++;
+    }
```
Comment (DevBastiS) [inline Week_2_Arrays/substitution/substitution.c:65]:
```suggestion
    }
    
```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_2_Arrays/substitution/substitution.c (hunk 15 above)
Comment (DevBastiS) [reply Week_2_Arrays/substitution/substitution.c:65]: Sill open

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_2_Arrays/caesar/caesar.c [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+{
+    if (argc != 2)
+    {
+        printf("Usage: ./caesar key");
+        return 1;
+    }
+
+    for (int i = 0; i < strlen (argv[1]); i++)
+    {
+        if (!isdigit(argv[1][i]))
```
Comment (DevBastiS) [inline Week_2_Arrays/caesar/caesar.c:17]: Always use {} notation

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_2_Arrays/caesar/caesar.c (hunk 16 above)
Comment (DevBastiS) [reply Week_2_Arrays/caesar/caesar.c:17]: Still open

**Review by DevBastiS — CHANGES_REQUESTED** [review]
Fix Merge Conflicts

**Review by DevBastiS — APPROVED** [review]

**Review by MKrips — APPROVED** [review]

**Pull request merged**